module3/lessons/fe_error_handling.md

```diff
@@ -0,0 +1,159 @@
+  - Simply checking that there is *something* there isn't enough. We need to check that the data is accurate and the correct data type (remember that input values always come back as strings!).
+
+**Whether things go right or wrong, we should give the user feedback.**
+  - Users should know that the action was successfully fulfilled.
+  - Users should know if something went wrong (whether it was user error or not).
+  - We should wait until after the Promise has successfully resolved from a fetch to confirm the outcome with the user. This is what we will talk about next!
+
+
+## Handling Fetch Errors
+
+### Set Up
+We're going to use the same Ideabox FE and BE that we've used several times in previous lessons.
+
+1. Re-open **[this react-ideabox FE](https://github.com/turingschool-examples/react-ideabox)** repo.  
+  a. Navigate to that directory  
+  b. Run `git fetch`  
+  c. Run `git checkout error-handling`  
+  d. Run `npm start`  
+2. You'll also need the **[ideabox-api BE Repo](https://github.com/turingschool-examples/ideabox-api)** up and running.  
+  a. Navigate to that directory 
+  b. Run `node server.js`
+3. Make a tab in Chrome for:  
+  a. `localhost:3000`: You should see the React app up and running here  
+  b. `http://localhost:3001/api/v1/ideas`: You should see your list of ideas here
+
+<section class="call-to-action">
+### Explore
+
+1. First, turn off the server. You should see a user-friendly error message in your Ideabox. Explore the React code and find the code that is responsible for making this happen.
+2. Now, turn the server back on. The error message should disappear and the cards should correctly populate. Now, try to submit an idea with a missing field (leave either the title or description blank). Pay attention to the following:
+  - What does the user see?
+  - Look at the console. Is there an error there? What is the status code?
+  - Go to `http://localhost:3001/api/v1/ideas`. Was that data POSTed?
+
+Discuss: Do you think the `.catch()` was fired in the second scenario? What is your evidence? Why do you think this is happening?
+</section>
+
+### Triggering the catch
+
+Hmm...it seems like not all errors trigger the `catch`. Which means we can't rely on the `catch` to let the user know if things have gone wrong. Let's dig into the response object. Let's add a console.log:
+
+```js
+  .then(response => {
+    console.log(response)
+    return response.json()
+  })
+```
+
+Now, let's compare that response object in two scenarios:
+
+1. The user fills everything out correctly:
+```js
+{
+  type: 'cors', 
+  url: 'http://localhost:3001/api/v1/ideas', 
+  redirected: false, 
+  status: 201, 
+  ok: true, 
+}
+```
+
+2. The user misses a field:
+```js
+{
+  type: 'cors', 
+  url: 'http://localhost:3001/api/v1/ideas', 
+  redirected: false, 
+  status: 422, 
+  ok: false, 
+}
+```
+
+Do we think we could use any of these properties to check if the network request was successful?
```

Comment:
I wonder if there is more we can add here to bridge the gap between using the .ok property to confirm whether the response was successful and then how to write code to handle an unsuccessful response if its not.

I'd love to have them dig into the MDN docs to understand the capital-E error before getting to the possible solution.